ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +45,445 @@ export function showErrorMessages(errors: any) {
+}
+
+function validateUsng(value: string) {
+  if (value === '') {
+    return { error: true, message: 'USNG / MGRS coordinates cannot be empty' }
+  }
+  const result = converter.USNGtoLL(value, true)
+  const isInvalid = Number.isNaN(result.lat) || Number.isNaN(result.lon)
+  return {
+    error: isInvalid,
+    message: isInvalid ? 'Invalid USNG / MGRS coordinates' : '',
+  }
+}
+
+function upsValidDistance(distance: number) {
+  return distance >= 800000 && distance <= 3200000
+}
+
+function validateUtmUps(key: string, value: any) {
+  let { utmUpsEasting, utmUpsNorthing, zoneNumber, hemisphere } = value
+  const northernHemisphere = hemisphere.toUpperCase() === 'NORTHERN'
+  zoneNumber = Number.parseInt(zoneNumber)
+  const isUps = zoneNumber === 0
+  let error = { error: false, message: '' }
+  // Number('') returns 0, so we can't just blindly cast to number
+  // since we want to differentiate '' from 0
+  let easting = utmUpsEasting === '' ? NaN : Number(utmUpsEasting)
+  let northing = utmUpsNorthing === '' ? NaN : Number(utmUpsNorthing)
+  const isNorthingInvalid = isNaN(northing) && utmUpsNorthing !== undefined
+  const isEastingInvalid = isNaN(easting) && utmUpsEasting !== undefined
+  if (!isNaN(easting)) {
+    easting = Number.parseFloat(utmUpsEasting)
+  } else if (
+    key === 'utmUpsEasting' &&
+    utmUpsEasting !== undefined &&
+    !isNorthingInvalid
+  ) {
+    return { error: true, message: 'Easting value is invalid' }
+  }
+  if (!isNaN(northing)) {
+    northing = Number.parseFloat(utmUpsNorthing)
+    northing =
+      isUps || northernHemisphere ? northing : northing - NORTHING_OFFSET
+  } else if (
+    key === 'utmUpsNorthing' &&
+    utmUpsNorthing !== undefined &&
+    !isEastingInvalid
+  ) {
+    return { error: true, message: 'Northing value is invalid' }
+  }
+  if (isUps && (!upsValidDistance(northing) || !upsValidDistance(easting))) {
+    return { error: true, message: 'Invalid UPS distance' }
+  }
+  const utmUpsParts = {
+    easting,
+    northing,
+    zoneNumber,
+    hemisphere,
+    northPole: northernHemisphere,
+  }
+  // These checks are to ensure that we only mark a value as "invalid"
+  // if the user has entered something already
+  let { lat, lon } = converter.UTMUPStoLL(utmUpsParts)
+  lon = lon % 360
+  if (lon < -180) {
+    lon = lon + 360
+  }
+  if (lon > 180) {
+    lon = lon - 360
+  }
+  // we want to validate using the hasPointError method, but only if they're both defined
+  // if one or more is undefined, we want to return true
+  const isLatLonValid =
+    !hasPointError([lon, lat]) ||
+    utmUpsNorthing === undefined ||
+    utmUpsEasting === undefined
+  if ((isNorthingInvalid && isEastingInvalid) || !isLatLonValid) {
+    return { error: true, message: 'Invalid UTM/UPS coordinates' }
+  }
+  return error
+}
+
+function validateRadiusLineBuffer(key: string, value: string) {
+  const label = key === 'lineWidth' ? 'Buffer ' : 'Radius '
```

Comment:
✏️ just to match the label of the text field
```suggestion
  const label = key === 'lineWidth' ? 'Buffer width' : 'Radius '
```